packages/strapi-connector-bookshelf/lib/mount-models.js

```diff
@@ -354,40 +354,55 @@ module.exports = async ({ models, target }, ctx, { selfFinalize = false } = {})
         }
         case 'belongsToMorph':
         case 'belongsToManyMorph': {
-          const association = definition.associations.find(
-            association => association.alias === name
-          );
+          const association = _.find(definition.associations, { alias: name });
 
-          const morphValues = association.related.map(id => {
-            let models = Object.values(strapi.models).filter(model => model.globalId === id);
+          const morphModelsAndFields = association.related.map(id => {
+            let modelFound = _.find(strapi.models, { globalId: id });
 
-            if (models.length === 0) {
-              models = Object.values(strapi.components).filter(model => model.globalId === id);
+            if (!modelFound) {
+              modelFound = _.find(strapi.components, { globalId: id });
             }
 
-            if (models.length === 0) {
-              models = Object.keys(strapi.plugins).reduce((acc, current) => {
-                const models = Object.values(strapi.plugins[current].models).filter(
-                  model => model.globalId === id
-                );
-
-                if (acc.length === 0 && models.length > 0) {
-                  acc = models;
-                }
-
-                return acc;
-              }, []);
+            if (!modelFound) {
+              _.forIn(strapi.plugins, plugin => {
+                modelFound = _.find(plugin.models, { globalId: id });
+                return !modelFound; // break
+              });
             }
 
-            if (models.length === 0) {
+            if (!modelFound) {
               strapi.log.error(`Impossible to register the '${model}' model.`);
               strapi.log.error('The collection name cannot be found for the morphTo method.');
               strapi.stop();
             }
 
-            return models[0].collectionName;
+            const relatedFields = _.reduce(
```

Comment:
do we realy need this added complexity ? Does bookshelf throw if a media attributes is removed ?

Reply:
Yes bookshelf throws if it cannot find the field become it has been removed :/